vm/systemSmartContracts/esdt.go

```diff
@@ -823,6 +919,78 @@ func (e *esdt) checkArgumentsForSpecialRoleChanges(args *vmcommon.ContractCallIn
 	return vmcommon.Ok
 }
 
+func isCreateNFTRoleInArgs(args [][]byte) bool {
+	for _, arg := range args {
+		if bytes.Equal(arg, []byte(core.ESDTRoleNFTCreate)) {
+			return true
+		}
+	}
+	return false
+}
+
+func checkIfCreateNFTRoleExists(args [][]byte, token *ESDTData) error {
```

Comment:
It is unclear what this method should do. 
1. No matter if ESDTRoleNFTCreate exists or NOT in special roles it returns nil if it NOT exists in args
2. If ESDTRoleNFTCreate exists in args it returns err if it also exists in special roles
3. If ESDTRoleNFTCreate exists in args it returns nil if it NOT exists in special roles

Reply:
yes. it should do exactly that. do you have a better name suggestion ? :)

Reply:
It checks if the role exists only in args? or only in token? or in both of them? When this method should return nil?

Reply:
first checks in args, and if there are no just return nil. If there is in arg, than it is checked in roles as well.